src/clients/UBAClient/UBAClientTypes.ts

```diff
@@ -0,0 +1,41 @@
+import { BigNumber } from "ethers";
+import { UbaFlow } from "../../interfaces";
+import { UBAActionType } from "../../UBAFeeCalculator/UBAFeeTypes";
+import UBAConfig from "../../UBAFeeCalculator/UBAFeeConfig";
+
+export type RequestValidReturnType = { valid: false; reason: string } | { valid: true };
+export type OpeningBalanceReturnType = { blockNumber: number; spokePoolBalance: BigNumber };
+export type BalancingFeeReturnType = { balancingFee: BigNumber; actionType: UBAActionType };
+export type SystemFeeResult = { lpFee: BigNumber; depositBalancingFee: BigNumber; systemFee: BigNumber };
+export type RelayerFeeResult = {
+  relayerGasFee: BigNumber;
+  relayerCapitalFee: BigNumber;
+  relayerBalancingFee: BigNumber;
+  relayerFee: BigNumber;
+  amountTooLow: boolean;
+};
+
+export type UBAChainState = {
+  spokeChain: {
+    deploymentBlockNumber: number;
+    bundleEndBlockNumber: number;
+    latestBlockNumber: number;
+  };
+  bundles: {
+    [tokenSymbol: string]: UBABundleState[];
+  };
+};
+
+export type UBABundleState = {
+  openingBalance: BigNumber;
+  blockNumber: number;
+  config: {
+    ubaConfig: UBAConfig;
+  };
+  flows: {
+    flow: UbaFlow;
+    systemFee: SystemFeeResult;
+    relayerFee: RelayerFeeResult;
+    runningBalance: BigNumber;
```

Comment:
nit: can you add `netRunningBalanceAdjustment` here too? We'll need this to construct `netSendAmounts` in the Dataworker and you should take into account when computing this `runningBalance` whether we need to make an adjustment due to thresholds

Reply:
Added